ert_data/measured.py

```diff
@@ -0,0 +1,129 @@
+        self._set_data(self._remove_inactive_observations())
+
+    def _remove_inactive_observations(self):
+        """Removes columns with one or more NaN values."""
+        filtered_dataset = self.data.dropna(axis=1)
+        if filtered_dataset.empty:
+            raise ValueError(
+                "This operation results in an empty dataset (could be due to one or more failed realizations)"
+            )
+        return filtered_dataset
+
+    def is_empty(self):
+        return self.data.empty
+
+    def _get_data(self, observation_keys, index_lists):
+        """
+        Adds simulated and observed data and returns a dataframe where ensamble
+        members will have a data key, observed data will be named OBS and
+        observed standard deviation will be named STD.
+        """
+        measured_data = pd.DataFrame()
+        case_name = self._facade.get_current_case_name()
+
+        if index_lists is None:
+            index_lists = [None] * len(observation_keys)
+
+        if len(index_lists) != len(observation_keys):
+            raise ValueError("index list must be same length as observations keys")
+
+        for key, index_list in zip(observation_keys, index_lists):
+            observation_type = self._facade.get_impl_type_name_for_obs_key(key)
+            data_loader = loader.data_loader_factory(observation_type)
+
+            data = data_loader(self._facade, key, case_name)
+            data = MeasuredData.filter_on_column_index(data, index_list)
+            data = pd.concat({key: data}, axis=1)
+
+            measured_data = pd.concat([measured_data, data], axis=1)
+
+        return measured_data
+
+    def filter_ensamble_std(self, std_cutoff):
+        self._set_data(self._filter_ensamble_std(std_cutoff))
+
+    def filter_ens_mean_obs(self, alpha):
+        self._set_data(self._filter_ens_mean_obs(alpha))
+
+    def _filter_ensamble_std(self, std_cutoff):
+        """
+        Filters on ensamble variation versus a user defined standard
+        deviation cutoff. If there is not enough variation in the measurements
+        the data point is removed.
+        """
+        ens_std = self.data[~self.data.index.isin(["OBS", "STD"])].std()
+        std_filter = (ens_std <= std_cutoff)
+        return self.data.drop(columns=std_filter[std_filter].index)
+
+    def _filter_ens_mean_obs(self, alpha):
+        """
+        Filters on distance between the observed data and the ensamble mean
+        based on variation and a user defined alpha.
+        """
+        ens_mean = self.data[~self.data.index.isin(["OBS", "STD"])].mean()
+        ens_std = self.data[~self.data.index.isin(["OBS", "STD"])].std()
+        obs_values = self.data.loc["OBS"]
+        obs_std = self.data.loc["STD"]
+
+        mean_filter = abs(obs_values - ens_mean) > alpha * (ens_std + obs_std)
+
+        return self.data.drop(columns=mean_filter[mean_filter].index)
+
+    @staticmethod
+    def filter_on_column_index(dataframe, index_list):
```

Comment:
This should probably be private no? Seems much to general to be exposed as part of the module...

Reply:
Fine by me, can expose it later should there be any use for it.